pkg/templates/pdbminavailable/template.go

```diff
@@ -0,0 +1,184 @@
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+const (
+	templateKey = "pdb-min-available"
+)
+
+func init() {
+	templates.Register(check.Template{
+		HumanName:   "Pod Disruption Budget Configuration Checks",
+		Key:         templateKey,
+		Description: "Flag non-optimal PDB configurations",
+		SupportedObjectKinds: config.ObjectKindsDesc{
+			ObjectKinds: []string{
+				objectkinds.PodDisruptionBudget},
+		},
+		Parameters:             params.ParamDescs,
+		ParseAndValidateParams: params.ParseAndValidate,
+		Instantiate: params.WrapInstantiateFunc(func(p params.Params) (check.Func, error) {
+			return func(lintCtx lintcontext.LintContext, object lintcontext.Object) []diagnostic.Diagnostic {
+
+				var results []diagnostic.Diagnostic
+
+				// Get the PDB provided
+				pdb, ok := object.K8sObject.(*pdbV1.PodDisruptionBudget)
+				if !ok {
+					return nil
+				}
+
+				// If MinAvailable isn't set, then no need to check
+				if pdb.Spec.MinAvailable == nil {
+					return results
+				}
+
+				// Extract the MinAvailable value from the spec and check if it's a number or percentage
+				value, isPercent, err := getIntOrPercentValueSafelyFromString(pdb.Spec.MinAvailable.String())
+				if err != nil {
+					return []diagnostic.Diagnostic{
+						{
+							Message: fmt.Sprintf("pdb has invalid MinAvailable value: %v", err),
+						},
+					}
+				}
+
+				// If the value is a percentage, handle the case where the MinValue is set to 100%
+				// as DeploymentLike's replica counts don't need to be compared
+				if isPercent {
+					if value == 100 {
+						return []diagnostic.Diagnostic{
+							{
+								Message: "PDB has minimum available replicas set to 100 percent of replicas",
+							},
+						}
+					}
+				}
+
+				// Build the label selector for the PDB to use for comparison
+				labelSelector, err := metaV1.LabelSelectorAsSelector(&metaV1.LabelSelector{
+					MatchLabels:      pdb.Spec.Selector.MatchLabels,
+					MatchExpressions: pdb.Spec.Selector.MatchExpressions},
+				)
+
+				if err != nil {
+					return []diagnostic.Diagnostic{
+						{
+							Message: fmt.Sprintf("pdb has invalid label selector: %v", err),
+						},
+					}
+				}
+
+				// Evaluate Deploymet Likes in the lintContext to see if they have MinAvailable set too low
+				deploymentLikes, err := getDeploymentLikeObjects(lintCtx, labelSelector, pdb.Namespace)
+				if err != nil {
+					return []diagnostic.Diagnostic{
+						{
+							Message: fmt.Sprintf("Failed to retrieve deployments matching the PDB's label selector within namespace %s: %v", pdb.Namespace, err.Error()),
+						},
+					}
+				}
+
+				for _, dl := range deploymentLikes {
+					pdbMinAvailable := value
+					replicas, _ := extract.Replicas(dl)
+					if isPercent {
+						// Calulate the actual value of the MinAvailable with respect to the Replica count if a percentage is set
+						pdbMinAvailable = int(math.Floor(float64(replicas) * (float64(value) / float64(100))))
```

Comment:
```suggestion
						pdbMinAvailable = int(replicas) * value / 100
```

Reply:
I already had this and was asked to change it to use float64s :)

Reply:
https://github.com/stackrox/kube-linter/pull/507#discussion_r1156852629